src/components/map/LayersService.js

```diff
@@ -730,33 +733,36 @@ goog.require('ga_urlutils_service');
                 preload: gaNetworkStatus.offline ? gaMapUtils.preload : 0,
                 useInterimTilesOnError: gaNetworkStatus.offline
               });
+              gaDefinePropertiesForLayer(olLayer);
             }
           } else if (config.type === 'aggregate') {
             var subLayersIds = config.subLayersIds;
             var i, len = subLayersIds.length;
-            var subLayers = new Array(len);
-            for (i = 0; i < len; i++) {
-              var subLayerConf = layers[subLayersIds[i]];
-              if (styleUrl && subLayerConf) {
-                layers[subLayersIds[i]].styleUrl = config.styleUrl;
+            var createSubLayers = function(olLayer, glStyle) {
+              var subLayers = new Array(len);
+              olLayer.glStyle = glStyle;
+              for (i = 0; i < len; i++) {
+                subLayers[i] = that.getOlLayerById(subLayersIds[i], {
+                  glStyle: glStyle
+                });
               }
-              subLayers[i] = this.getOlLayerById(subLayersIds[i], opts);
-            }
-
+              olLayer.setLayers(new ol.Collection(subLayers));
+            };
             olLayer = new ol.layer.Group({
               minResolution: config.minResolution,
               maxResolution: config.maxResolution,
               opacity: config.opacity || 1
             });
-            if (styleUrl) {
-              gaGlStyle.get(styleUrl).then((glStyle) => {
-                gaMapUtils.applyGlStyleToOlLayer(olLayer, glStyle);
-                olLayer.setLayers(new ol.Collection(subLayers));
+            gaDefinePropertiesForLayer(olLayer);
+            if (glStyle) {
+              createSubLayers(olLayer, glStyle);
+            } else if (styleUrl) {
```

Comment:
Same here. I propose a small refactoring , it will avoid to create a function before, help to understand better the code:
``` 
var p =  (glStyle) ? $q.when(glStyle) : gaGlStyle.get(styleUrl);
p.then(function(glStyle) {
￼    var subLayers = new Array(len);
    olLayer.glStyle = glStyle;
    for (i = 0; i < len; i++) { ....
```

Reply:
evrything is a promise!